proxies/live/apiproxy/resources/jsc/SetErrorVars.js

```diff
@@ -0,0 +1,210 @@
+        title = "Property cannot be null"
+        detail = "The property at the specified location is required, but a null value was passed in the request."
+        errors.push(createErrorObject(title, detail, pointer))
+      }
+
+      // $.data.attributes.routingPlanId
+      pointer = "/data/attributes/routingPlanId"
+      if (data.attributes.routingPlanId) {
+        if (typeof data.attributes.routingPlanId !== "string" && !uuidRegex.test(data.attributes.routingPlanId)) {
+          title = "Invalid value"
+          detail = "The property at the specified location does not allow this value."
+          errors.push(createErrorObject(title, detail, pointer))
+        }
+      } else {
+        title = "Missing property"
+        detail = "The property at the specified location is required, but was not present in the request."
+        errors.push(createErrorObject(title, detail, pointer))
+      }
+
+      // $.data.attributes.batchReference
+      pointer = "/data/attributes/batchReference"
+      if (data.attributes.batchReference) {
+        if (typeof data.attributes.batchReference !== "string" && !uuidRegex.test(data.attributes.batchReference)) {
+          title = "Invalid value"
+          detail = "The property at the specified location does not allow this value."
+          errors.push(createErrorObject(title, detail, pointer))
+        }
+      } else {
+        title = "Missing property"
+        detail = "The property at the specified location is required, but was not present in the request."
+        errors.push(createErrorObject(title, detail, pointer))
+      }
+
+      // $.data.attributes.recipients
+      pointer = "/data/attributes/recipients"
+      if (data.attributes.recipients) {
+        if (!typeof data.attributes.recipients.isArray) {
+          title = "Invalid value"
+          detail = "The property at the specified location does not allow this value."
+          errors.push(createErrorObject(title, detail, pointer))
+        } else {
+          if (data.attributes.recipients.length() < 1) {
+            title = "Too few items"
+            detail = "The property at the specified location contains too few items."
+            errors.push(createErrorObject(title, detail, pointer))
+          } else {
+
+            // $.data.attributes.recipients.x
+            for (var i = 0; i < data.attributes.recipients.length(); i++) {
+              pointer = "/data/attributes/recipients/" + i
+              var item = data.attributes.recipients[i]
+              if (!item || item === null || item === undefined) {
+                title = "Missing property"
+                detail = "The property at the specified location is required, but was not present in the request."
+                errors.push(createErrorObject(title, detail, pointer))
+              } else {
+
+                // $.data.attributes.recipients.x.recipientReference
+                pointer = "/data/attributes/recipients/" + i + "/recipientReference"
```

Comment:
Interpolation instead:

`/data/attributes/recipients/${i}/recipientReference`

Reply:
Yeah unfortunately the javascript engine in Apigee doesn't support some features, so we're stuck with basic string concatenation